src/com/phpuaca/completion/MockedClassTypeProvider.java

```diff
@@ -0,0 +1,232 @@
+        }
+
+        List<PhpMethodSignature> signatures = getSignatures();
+        if (signatures.size() == 0) {
+            return null;
+        }
+
+        MethodReference methodReference = (MethodReference) element;
+        String refSignature = methodReference.getSignature();
+
+        if (StringUtil.isEmpty(refSignature)) {
+            return null;
+        }
+
+        // Get signature from:  $this->getMockBuilder('CLASS')->getMock()
+        MethodReference mockBuilderMethodReference = (new PhpMethodChain(methodReference)).findMethodReference("getMockBuilder");
+
+        if (mockBuilderMethodReference != null && "getMock".equals(methodReference.getName())) {
+            PsiElement[] mockBuilderParams = mockBuilderMethodReference.getParameters();
+            String signature = buildParameterMethodSignature(refSignature, 0, mockBuilderParams);
+
+            if (signature != null) {
+                return signature;
+            }
+        }
+
+        // Set MethodProphecy signature for:  $prophecy->method()
+        PsiElement child = methodReference.getFirstChild();
+        if (child instanceof Variable) {
+            String expression = ((Variable)child).getType().toString();
+            // get back our original call
+            int endIndex = expression.lastIndexOf(TRIM_KEY);
+            if(endIndex != -1) {
+                PhpIndex phpIndex = PhpIndex.getInstance(element.getProject());
+
+                MethodReference prophesizeMethod = (new PhpVariable((Variable)child)).findClosestAssignment();
+
+                if (prophesizeMethod != null) {
+                    PhpNamedElement prophCall = phpIndex.getBySignature(prophesizeMethod.getSignature()).iterator().next();
+
+                    if(prophCall != null && prophCall.getFQN() != null &&
+                            (prophCall.getFQN().equals("\\PHPUnit_Framework_TestCase.prophesize") || prophCall.getFQN().equals("\\Prophecy\\Prophet.prophesize"))) {
+                        String prophType = prophCall.getType().toString();
+                        if(prophType.equals("\\Prophecy\\Prophecy\\ObjectProphecy")) {
+                            if("reveal".equals(methodReference.getName())) {
+                                return buildParameterMethodSignature(refSignature, 0, prophesizeMethod.getParameters());
+                            } else {
+                                // check if methodReference is not in ObjectProphecy (it is a mocked method)
+                                PhpMethodResolver resolver = new PhpMethodResolver(methodReference);
+                                if(resolver.resolve()) {
+                                    String resolvedClass = resolver.getResolvedClass().getFQN();
+                                    if(resolvedClass != null && !resolvedClass.equals("\\Prophecy\\Prophecy\\ObjectProphecy")) {
+                                        return "\\Prophecy\\Prophecy\\MethodProphecy";
+                                    }
```

Comment:
Yeah, sorry it get's a little bit special here, because of the handling of method calls from `ObjectProphecy` objects.

You want me to refactor it?